### PR TITLE
TST: add workflow_dispatch for pysat RC

### PR DESCRIPTION
**Pull request by jklenzing**
# Description

Addresses pysat/pysat#1042

Adds a dispatch workflow to test this package with the latest pysat RC.

## Type of change

- New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Inspecting the logs at https://github.com/pysat/pysatMissions/actions/runs/3439753149

Ensure that pysat is installed only from the test pypi server.

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: .github/workflows/pysat_rc.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+name: Test with latest pysat RC
+
+on: [workflow_dispatch]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
```
Comment (aburrell) [inline .github/workflows/pysat_rc.yml:13]: Not mac?

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ This project adheres to [Semantic Versioning](https://semver.org/).
 ## [0.X.X] - 2022-XX-XX
 * Maintenance
   * Update pytest syntax
+  * Add manual GA tests for pysat RC
```
Comment (aburrell) [inline CHANGELOG.md:8]:
```suggestion
  * Add manual GitHub Actions tests for pysat RC
```

File: .github/workflows/pysat_rc.yml (hunk 1 above)
Comment (jklenzing) [reply .github/workflows/pysat_rc.yml:13]: not in the main jobs yet

Comment (aburrell) [reply .github/workflows/pysat_rc.yml:13]: Might be worth adding a comment with a TODO?

Comment (jklenzing) [reply .github/workflows/pysat_rc.yml:13]: Added #101 with this and other things we are not currently testing in main due to fortran issues

**Review by aburrell — APPROVED** [review]

**Pull request merged**